app/uk/gov/hmrc/cgtpropertydisposals/repositories/returns/DraftReturnsRepository.scala

```diff
@@ -76,6 +78,24 @@ class DefaultDraftReturnsRepository @Inject() (component: ReactiveMongoComponent
   override def save(draftReturn: DraftReturn, cgtReference: CgtReference): EitherT[Future, Error, Unit] =
     EitherT(set(draftReturn.id.toString, DraftReturnWithCgtReference(draftReturn, cgtReference, draftReturn.id)))
 
+  override def delete(cgtReference: CgtReference): EitherT[Future, Error, Unit] =
+    EitherT[Future, Error, Unit](
+      remove("return.cgtReference" -> cgtReference.value)
```

Comment:
will this delete all draft returns for that cgt reference? You could write a test for it like this to double check: https://github.com/hmrc/cgt-property-disposals/blob/master/test/uk/gov/hmrc/cgtpropertydisposals/repositories/returns/DraftReturnsRepositorySpec.scala#L68

Reply:
it would not, I fixed query. I will add a test as you suggested. Thx